src/classes/Contacts.cls

```diff
@@ -0,0 +1,732 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Domain class for Contacts in NPSP.
+*/
+public inherited sharing class Contacts extends fflib_SObjects {
```

Comment:
Overall I like how the class encapsulates Contact related logic, however I have a few thoughts:

- It's a huge class and it may be worth considering abstracting common logic out into its own service classes - naming, address, etc.
- I see your point on not adding block comments, but there are a lot of methods here that could use comments. We've always settled that every method requires a block comment, and I'm not sure we should try to change that rule here. Yes, some comments on some methods may be redundant, but I think there are enough complex methods in here that commenting will be beneficial.
- There's a lot of logic that [I believe] is unique to the older account models (Individual and Bucket). What would you think of abstracting those out into another class or even an inner class - as a way of making it super clear while reading the code that these are methods in support of legacy models?

Reply:
* Yes, this is a huge class.  Since we are rolling this WI over to next Sprint, I will start a next round of refactor and see if there is more logic that we can abstract out to reduce its size.  Overall, this will be an ongoing process.  While we might not get all the way to our perfect refactoring point in this work, we are moving in the right direction, and we should begin to regularly update our domain model and refactor code when the two get out of line: https://lucid.app/lucidchart/d10afa5b-bbc2-443f-9817-619c148235c5/edit?page=0_0#?folder_id=home&browser=icon
* We should definitely not add block comments to every method.  NPSP needs to move away from that rule in my opinion.  Every time we think "this method does multiple things and needs a comment" that is a sign that the method needs to be refactored.  We want methods that are single chunks of code that do one thing and have expressive names. I'll do another pass and continue to try to break down the methods that do multiple things.  
* Splitting out classes by Account model type is an interesting idea.  I'll look and see if I can start that process.